tls/s2n_client_hello.c

```diff
@@ -195,6 +195,8 @@ static int s2n_parse_client_hello(struct s2n_connection *conn)
     /* This is going to be our default if the client has no preference. */
     conn->secure.server_ecc_evp_params.negotiated_curve = ecc_pref->ecc_curves[0];
 
+    ENSURE_POSIX(conn->secure.server_ecc_evp_params.evp_pkey == NULL, S2N_ERR_BAD_MESSAGE);
```

Comment:
Why are we making this check here? Shouldn't the key share extension own it?

Reply:
Technically yes, I was putting a defensive check to catch the error.